### PR TITLE
Support pinch to zoom/rotate in modal image

### PR DESCRIPTION
**Pull request by fourlastor**
This PR adds support for pinch to zoom and rotate (Android lags because it can't handle both recording and using the emulator at the same time, apparently):

| Android | iOS |
| --- | --- |
|device-2018-01-16-113334 | video-ios |

Gestures is ignored instead of being tested because I don't know if it's even possible to test both gestures and animated values.

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/b9a0ecfca8611d17

**Comment by colincclark** [comment]
What JIRA ticket does this relate to? I am a bit perplexed as to why this was a requirement...

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/2ecf6656ba6235dc

**Comment by fourlastor** [comment]
@colincclark `REPLAT-1181`, my bad, it's only on the branch name, not very obvious

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://www.bitrise.io/build/91310fad6ac94083

**Comment by colincclark** [comment]
@fourlastor I dont see anything about rotating the image in the JIRA ticket. For simple pinch gesture functionality, there is `ScrollView`. See https://facebook.github.io/react-native/docs/scrollview.html#maximumzoomscale - might be simpler to implement, unless I am wrong about the requirement to turn the image...

**Comment by fourlastor** [comment]
@colincclark gave the ScrollView a quick try, there are a few problems that imho don't make it a good candidate for this:

1. It works only on iOS
2. The zoom doesn't reset when you lift the fingers (might have not seen it on the docs)
3. The zoom is centred on the top right corner

File: packages/image/gestures.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    });
+  }
+
+  handlePinchStart({ nativeEvent: { touches } }) {
+    this.startDistance = distanceBetweenTouches(touches);
+    this.startAngle = angleBetweenTouches(touches);
+  }
+
+  handlePinchChange({ nativeEvent: { touches } }) {
+    if (touches.length < 2) {
```
Comment (craigbilner) [inline packages/image/gestures.js]: what should happen if greater than 2?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/eca5681243423ad317fff00c260a0c0ab57e7cbc/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/3c6098e94d01af1fda2145e68388c62b627c7df5/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/5e65e91fca7fb8f013d6a17f174b6e28e787d73a/dextrose-test/index.html

File: packages/image/__tests__/android/__snapshots__/image.android.test.js.snap [hunk 2]
```diff
@@ -1,5 +1,161 @@
...
+exports[`Image test on android Gestures should rotate clockwise when turning fingers 1`] = `
+<AnimatedComponent
+  style={
+    Object {
+      "transform": Array [
+        Object {
+          "scale": 1,
+        },
+        Object {
+          "rotate": "270 deg",
```
Comment (craigbilner) [inline packages/image/__tests__/android/__snapshots__/image.android.test.js.snap]: @fourlastor can't recreate on emulator, need to try on a device

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/bd3ced3323d68a0307530a8b57c0b2de26f4577c/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/b18816b813d7e8d97e865189471d32cc39f20752/dextrose-test/index.html

**Pull request merged**